src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/CollectionDataContract.cs

```diff
@@ -68,14 +68,16 @@ public T Value
 
     [DataContract(Namespace = "http://schemas.microsoft.com/2003/10/Serialization/Arrays")]
 #if USE_REFEMIT
-    public struct KeyValue<K, V>
+    public class KeyValue<K, V>
 #else
```

Comment:
I forgot to mention that this change is for NetCore only since this is the part where implementation on Desktop and NetCore diverges. In Desktop we use RuntimeServices.GetUninitlaizedObject to create new object instance whereas in NetCore that API is not available. We opt for getting that method through reflection and this perf issue is incurred.